packages/patternfly-4/react-table/src/components/Table/utils/decorators/selectable.js

```diff
@@ -3,27 +3,40 @@ import { css } from '@patternfly/react-styles';
 import { tableCheck } from '@patternfly/patternfly-next/components/Table/table.css';
 import SelectColumn from '../../SelectColumn';
 
-export default (label, { column: { extraParams: { onSelect, rowLabeledBy = 'simple-node' } }, rowIndex, rowData }) => {
+export default (
+  label,
+  {
+    column: {
+      extraParams: { onSelect, rowLabeledBy = 'simple-node' }
+    },
+    rowIndex,
+    rowData
+  }
+) => {
   const rowId = rowIndex !== undefined ? rowIndex : -1;
   function selectClick(event) {
-    let selected = rowIndex === undefined ? event.target.checked : rowData && !rowData.selected;
-    onSelect && onSelect(selected, selected, rowId);
+    const selected = rowIndex === undefined ? event.target.checked : rowData && !rowData.selected;
+    onSelect && onSelect(event, selected, rowId, rowData);
```

Comment:
rowData is needed to access the `row.id` in cases of a selectable virtualized table

Reply:
hey @karelhala - i think this one will be needed sooner than later. I was testing `onSelect` callbacks w/ @TheRealJon today and it looks like we'll definitely need `rowData` returned in order to appropriately select by an OpenShift identifier. I'm happy to open another issue to target that specifically... and resolve in a separate PR if you'd like.

This piece can be targeted for this:
https://github.com/patternfly/patternfly-react/blob/master/packages/patternfly-4/react-table/src/components/Table/utils/decorators/selectable.js#L27

it looks like this change was not included in #1227 either...

thoughts?

Reply:
+1 I think it is better to send rowData anywhere it might be remotely useful

Reply:
opened #1472